tests/Benchmarks/System.Text.Formatting/StringFormatterPerf.cs

```diff
@@ -0,0 +1,241 @@
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using System.Text.Formatting;
+
+namespace System.Text.Formatting.Benchmarks
+{
+    public class StringFormatterPerf
+    {
+        [Params(10, 1000)]
+        public int NumbersToWrite { get; set; }
+        private static ArrayPool<byte> pool = ArrayPool<byte>.Shared;
+
+        [Benchmark]
+        public void InvariantFormatIntDec()
+        {
+            var sb = new StringFormatter(NumbersToWrite, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntDecClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntHex()
+        {
+            var sb = new StringFormatter(NumbersToWrite, pool);
+            var format = new StandardFormat('X', StandardFormat.NoPrecision);
+
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)), format);
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatIntHexClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)).ToString("X"));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatStruct()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 2, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void FormatGuid()
+        {
+            var guid = Guid.NewGuid();
+            var guidsToWrite = NumbersToWrite / 10;
+
+            var sb = new StringFormatter(guidsToWrite * 36, pool);
+            for (int i = 0; i < guidsToWrite; i++)
+            {
+                sb.Append(guid);
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void InvariantFormatStructClr()
+        {
+            var sb = new StringBuilder(NumbersToWrite * 2);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(new Age(i % 10));
+            }
+
+            var text = sb.ToString();
+        }
+
+        [Benchmark]
+        public void CustomCultureFormat()
+        {
+            var sb = new StringFormatter(NumbersToWrite * 3, pool);
+            sb.SymbolTable = CreateCustomCulture();
```

Comment:
`CreateCustomCulture()` can affect the final result, it would be nice to move it to a targeted setup method. sth like

`[GlobalSetup(Target = nameof(CustomCultureFormat))]`